aptos-move/aptos-native-interface/src/builder.rs

```diff
@@ -27,10 +28,10 @@ struct SharedData {
 /// Factory object that allows one to build native functions with ease.
 ///
 /// This enables native functions to access shared data, and interface with `SafeNativeContext`.
-#[derive(Debug)]
 pub struct SafeNativeBuilder {
     data: Arc<SharedData>,
     enable_incremental_gas_charging: bool,
+    gas_hook: Option<Arc<dyn Fn(DynamicExpression) + Send + Sync>>,
```

Comment:
I believe it's because in `executor.rs` there are shared vectors that is being used to capture the gas expression then record its abstract gas usage to write back into the original vector.

Reply:
I don't remember the exact details, but this stems from Move VM's (unnecessarily strict) requirement to put natives in `Arc`s. Would love to ease this requirement, but it would require some work, so let's do it separately.